src/MSALWrapper/AuthFlow/Broker.cs

```diff
@@ -84,58 +84,47 @@ protected override async Task<(TokenResult, IList<Exception>)> GetTokenInnerAsyn
         {
             IAccount account = await this.pcaWrapper.TryToGetCachedAccountAsync(this.preferredDomain)
                  ?? PublicClientApplication.OperatingSystemAccount;
-            this.logger.LogDebug($"Using cached account '{account?.Username}'");
 
+            TokenResult tokenResult = null;
             try
             {
-                try
+                tokenResult = await CachedAuth.TryCachedAuthAsync(
+                    this.logger,
+                    this.silentAuthTimeout,
+                    this.scopes,
+                    account,
+                    this.pcaWrapper,
+                    this.errors);
+
+                if (tokenResult == null)

```

Comment:
You could early return here and avoid nesting the giant `try` block below that way. Just something to consider.

Reply:
Will address in the follow up to not have to return the tuple.